pyquil/paulis.py

```diff
@@ -60,14 +60,15 @@ def __init__(self, op, index, coefficient=1.0):
         self._ops = {}
         if op != "I":
             self._ops[index] = op
-
-        self.coefficient = coefficient
+        if not isinstance(coefficient, Number):
+            raise ValueError("coefficient of PauliTerm must be a Number.")
+        self.coefficient = complex(coefficient)
```

Comment:
we should add a comment why we need to explicitly cast to a python `complex` type.